### PR TITLE
[luabinding] make CCAssert compatible Lua

### PR DESCRIPTION
**Pull request by dualface**
This patch avoid CCAssert interrupt program execution, and push an error message to the Lua Stack.

How to use:

``` lua
function __G__TRACKBACK__(errorMessage)
    CCLuaLog("----------------------------------------")
    CCLuaLog("LUA ERROR: "..tostring(errorMessage).."\n")
    CCLuaLog(debug.traceback("", 2))
    CCLuaLog("----------------------------------------")
end

xpcall(function()
    local scene = CCScene:create()
    local sprite = CCSprite:create("hello.png")

    scene:addChild(sprite)
    scene:addChild(sprite) -- ERROR: repeat addChild()
end, __G__TRACKBACK__)
```

output:

```
----------------------------------------
LUA ERROR: ASSERT FAILED ON LUA EXECUTE: child already added. It can't be added again


stack traceback:
    [C]: in function 'addChild'
    ...ce/Documents/Games/test/client_scripts/main_editor.lua:13: in function <...ce/Documents/Games/test/client_scripts/main_editor.lua:8>
    [C]: in function 'xpcall'
    ...ce/Documents/Games/test/client_scripts/main_editor.lua:8: in main chunk
----------------------------------------

```

### CODE REVIEW

File: cocos2dx/include/ccMacros.h [hunk 1]
```diff
@@ -35,7 +35,12 @@ of this software and associated documentation files (the "Software"), to deal
 #include "CCStdC.h"
 
 #ifndef CCAssert
+#if CC_LUA_ENGINE_ENABLED > 0
```
Comment (minggo) [inline cocos2dx/include/ccMacros.h]: Now we don't use macro to control lua related codes. Is it only used here?

Comment (dualface) [inline cocos2dx/include/ccMacros.h]: Now how to control lua related codes ?

Comment (minggo) [inline cocos2dx/include/ccMacros.h]: You can checkout the implementation of CCNode::cleanup().
Why can not just use CCAssert?

Comment (dualface) [inline cocos2dx/include/ccMacros.h]: CCAssert会中断执行。从Lua里调用某个cocos2dx方法时如果断言失败，开发者无法得到详细的错误信息。这个补丁让Lua引擎有机会捕获断言失败的情况，从而输出Lua调用栈，方便开发者找出问题是何处的lua代码引起的。

发自我的 iPhone

在 2012-12-24，14:36，minggo notifications@github.com 写道：

**Comment by dualface** [comment]
remove CC_LUA_ENGINE_ENABLED macro

**Pull request merged**